task-02

denisgubin opened the pull request:
task 02-sort-strings
task-02-pick
task-02-omit

denisgubin committed:
task 02-sort-strings
task-02-pick
task-02-omit

jsru commented:
Добавляю преподавателя (@ufocoder) для код-ревью.

ufocoder commented on 02-javascript-data-types/1-sort-strings/index.js:
```diff
@@ -4,6 +4,39 @@
+  let copyArray = [...arr];
+  let firstCharacterOfFirstElem = copyArray[0].charAt(0);
+
+  let locale = getLocale(firstCharacterOfFirstElem);
```
Нет необходимости в определении, какой язык использовать, можно указать несколько

```
strA.localeCompare(strB, ['ru', 'en'], { sensitivity: 'case' })
```
И тогда можно будет избавиться от вспомогательных функций и кода

ufocoder commented on 02-javascript-data-types/2-pick/index.js:
```diff
@@ -5,5 +5,37 @@
+  // В чем преимущество способа ниже с Object.entries(obj) перед текущим способом перебора полей объекта ?
+
+  for (const [key, value] of Object.entries(obj)) {
+    if (checkInList(key, ...fields)) {
```
Можно использовать `fields.includes(key)` вместо `checkInList` здесь и в следующем задании

ufocoder commented on 02-javascript-data-types/2-pick/index.js:
```diff
@@ -5,5 +5,37 @@
+  //     cloneObject[key] = obj[key];
+  //   }
+  // }
+  // В чем преимущество способа ниже с Object.entries(obj) перед текущим способом перебора полей объекта ?
```
Выберите один из способов реализации

ufocoder replied:
Объяснение о переборе будет на следующей встрече

denisgubin committed:
task 02-sort-strings
task-02-pick
task-02-omit

избавление от лишнего кода ниже:

let regExpEn = /[a-zA-z]/;
let regExpRu = /[а-яА-ЯЁё]/;

  let firstCharacterOfFirstElem = copyArray[0].charAt(0);

  let locale = getLocale(firstCharacterOfFirstElem);
  if (locale === "") {
    return arr;
  }

function getLocale(word) {
  let matchEn = regExpEn.test(word);
  if (matchEn === true) {
    return "en-US"; // determined 'en-US' locale
  }

  let matchRu = regExpRu.test(word);
  if (matchRu === true) {
    return "ru-RU"; // determined 'ru-RU' locale
  }

  return "";
}

/**
 * checkInList - Checking either value exists in fields
 * @param {string} value - value to find
 * @param {...string} fields - properties
 * @returns {boolean} - true if value exists in fields else false
 */
export function checkInList(value, ...fields) {
  for (let field of fields) {
    if (value === field) {
      return true;
    }
  }

  return false;
}

jsru commented:
Добавляю преподавателя (@ufocoder) для код-ревью.

denisgubin committed:
task 02-sort-strings
task-02-pick
task-02-omit

избавление от лишнего кода ниже:

/**
 * checkInList - Checking either value exists in fields
 * @param {string} value - value to find
 * @param {...string} fields - properties
 * @returns {boolean} - true if value exists in fields else false
 */
export function checkInList(value, ...fields) {
  for (let field of fields) {
    if (value === field) {
      return true;
    }
  }

  return false;
}

jsru commented:
Решение было обновлено, посмотрим что скажет @ufocoder

ufocoder commented on 02-javascript-data-types/1-sort-strings/index.js:
```diff
@@ -4,6 +4,20 @@
 
+  switch (param) {
+  case "desc":
+    return copyArray.sort((a, b) => b.localeCompare(a, [localeRu + "-u-kf-upper", localeEn + "-u-kf-upper"], {sensitivity: 'case'}));
```
Для повышения читаемости вынести функции сравнения за функцию `sortStrings` и можно не использовать switch/case, по умолчанию используйте одну из сортировок, которую можно определить по тестам или по значению по умолчанию

denisgubin committed:
task02 задача 01
Правки по функции sortStrings

denisgubin committed:
task02 задача 01
Правки по функции sortStrings

denisgubin committed:
task02 задача 01
Вернулся к старому решению

denisgubin committed:
task02 задача 01
Опять новое решение
